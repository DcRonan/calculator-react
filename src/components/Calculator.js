const Calculator = () => <div style={{ fontSize: '20px' }}>Let`s do some math!</div>;

export default Calculator;
