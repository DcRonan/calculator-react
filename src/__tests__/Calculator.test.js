import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('checks if Calculator component renders correctly', () => {
  test('should renders correct text', () => {
    render(<Calculator />);
    const text = screen.getByText(/Let`s do some math!/i);
    expect(text).toBeInTheDocument();
  });

  test('should render correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
