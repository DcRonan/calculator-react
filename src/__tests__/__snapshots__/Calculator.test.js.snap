// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if Calculator component renders correctly should render correctly 1`] = `
<div
  style={
    Object {
      "fontSize": "20px",
    }
  }
>
  Let\`s do some math!
</div>
`;
